src/frontends/paddle/src/op/sum.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs sum(const NodeContext& node) {
+    auto data = node.get_ng_inputs("X");
+    const auto input_type = data[0].get_element_type();
+    const auto shape = data[0].get_shape();
+    auto sum = data[0];
+    int i = 1;
+    for(; i<data.size(); i++) {
+        sum = std::make_shared<default_opset::Add>(sum, data[i]);
+    }
+    const auto temp = default_opset::Constant::create(input_type, {shape}, {0});
+    return node.default_single_output_mapping({std::make_shared<default_opset::Add>(sum, temp)}, {"Out"});
```

Comment:
Hi @Patrick-Star125 can we reuse the 'sum' here, not create a constant to build a new node again ?

Reply:
I thought so at first.
But if I reuse the `sum`, it will cause a error in the compile phase.
![image](https://user-images.githubusercontent.com/69072522/185854613-fee58b7c-8acc-407f-825e-3b7fad7c648d.png)

Reply:
Maybe we can create a 'Constant Node' to initialize the sum node ? but not initialize it by data[0] ? I guess.

Reply:
I thought so at first, so I tried to implement it as below.
![image](https://user-images.githubusercontent.com/69072522/185871512-d072e85b-5207-4617-9e50-be3daf8154ba.png)
But it will cause another error in the compile phase.
![image](https://user-images.githubusercontent.com/69072522/185871264-6a8fff7b-ac70-4bd0-ba6b-890d1ba5292a.png)

Reply:
> I thought so at first, so I tried to implement it as below. ![image](https://user-images.githubusercontent.com/69072522/185871512-d072e85b-5207-4617-9e50-be3daf8154ba.png) But it will cause another error in the compile phase. ![image](https://user-images.githubusercontent.com/69072522/185871264-6a8fff7b-ac70-4bd0-ba6b-890d1ba5292a.png)

Got it. Thanks. How about leaving the {std::make_shared<default_opset::Add>(sum, temp)} before the loop, to transfer the first input element into a Add node. And let's see other reviewer's comments.

Reply:
Thanks, the problem was solved by initializing the `sum` with a copied pointer of the `data` object.